students/shawn_degraw/session04/trigrams.py

```diff
@@ -0,0 +1,68 @@
+import random, string
+
+
+def build_trigrams(fileinput):
+    """
+    build up the trigrams dict from the list of words
+
+    returns a dict with:
+       keys: word pairs
+       values: list of followers
+    """
+    trigrams = {}
+    cleaning = str.maketrans('','',string.punctuation)
+
+    #but no error handling yet
+    with open(fileinput, 'r') as filehandle:
+        #read file, clean newlines and puncuation, make everything lower case for more word matching, put into list
+        words = (((filehandle.read().replace('\n', ' ')).translate(cleaning)).lower()).split()
+        
+        #populate trigrams dictionary
+        for i in range(len(words)-2):
+            if tuple(words[i:i+2]) in trigrams:
+                trigrams[tuple(words[i:i+2])].append(words[i + 2]) 
+            else: 
+                trigrams[tuple(words[i:i+2])] = [words[i + 2]]
```

Comment:
Minor: With `trigrams = collections.defaultdict(list)` (after `import collections` at the top), lines 22, 24, 25 can be safely removed. No need to change.